File: ruby/rna-transcription/rna_transcription.rb
class Complement
    def self.of_dna(dna)
        if dna.match(/[^ACGTU]/) != nil
            return ""
        end

        dna = dna.split("").map { |strand| 
            if strand == "G"
                strand = "C"
            elsif strand == "C"
                strand = "G"
            elsif strand == "T"
                strand = "A"
            elsif strand == "A"
                strand = "U"
            end
        }.join("")
        dna
    end
end

module BookKeeping
  VERSION = 4 # Where the version number matches the one in the test.
end

File: javascript/rna-transcription/rna-transcription.js
const DnaTranscriber = function () {};

DnaTranscriber.prototype.toRna = function(dna){
  return dna.split("").map(function(strand){
    switch(strand) {
      case "G":
        return strand = "C"
      case "C":
        return strand = "G"
      case "T":
        return strand = "A"
      case "A":
        return strand = "U"
      default:
        return strand = ""
    }
  }).join("");
}

module.exports = DnaTranscriber;